### PR TITLE
Added benchmark construction

### PR DESCRIPTION
**Pull request by KennethEnevoldsen**
This PR adds an outline for benchmark construction.

This is mostly to get an idea for task selection / benchmark construction. (see more at #837) and whether people agree with this approach.

Naturally, once a workflow is set up for creating a benchmark it is quite easy to add new ones so it might also be argued that we should just add a benchmark for all languages we believe is well enough covered. Let me know what you guys think.

### CODE REVIEW

**Comment by isaac-chung** [comment]
I think this addresses #368 , #367 , #365 , and #361 as well. Can mention those in the description if that's agreed.

**Comment by KennethEnevoldsen** [comment]
> I think this addresses https://github.com/embeddings-benchmark/mteb/issues/368 , https://github.com/embeddings-benchmark/mteb/issues/367 , https://github.com/embeddings-benchmark/mteb/issues/365 , and https://github.com/embeddings-benchmark/mteb/issues/361 as well. Can mention those in the description if that's agreed.


That is a great idea. We haven't added the lingua franca benchmark but should be easy enough.

File: mteb/tasks/Classification/zho/CMTEBClassification.py [hunk 1]
```diff
@@ -220,21 +220,7 @@ class JDReview(AbsTaskClassification):
...
-}
-""",
+        bibtex_citation="",  # missing
```
Comment (Muennighoff) [inline mteb/tasks/Classification/zho/CMTEBClassification.py:223]: Maybe it should cite the paper that created CMTEB (https://arxiv.org/abs/2309.07597) 🤔

**Review by Muennighoff — COMMENTED** [review]
Nice should this file eventually be merged into https://github.com/embeddings-benchmark/mteb/blob/main/mteb/benchmarks.py ?

**Comment by KennethEnevoldsen** [comment]
> Nice should this file eventually be merged into https://github.com/embeddings-benchmark/mteb/blob/main/mteb/benchmarks.py ?

Once we have downsampled the list of tasks - then yes

File: scripts/mmteb/task_selection/benchmark_construction.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+task_lists = {
+    "MTEB(eng, lite)": {
+        "tasks_to_search": MTEB_MAIN_EN,  # tasks to select from
+        "minimally_include": [],  # tasks that must be included in the final benchmark (considered high-quality, representative etc.)
+        "task_types_to_include": [  # task types that should be represented in the final benchmark
+            "Classification",
+            "Clustering",
+            "PairClassification",
+            "Reranking",
+            "Retrieval",
```
Comment (isaac-chung) [inline scripts/mmteb/task_selection/benchmark_construction.py:60]: I might have missed the discussion on this. Was the exclusion of STS and other tasks intentional?

Comment (KennethEnevoldsen) [reply scripts/mmteb/task_selection/benchmark_construction.py:60]: No def. Atm they are only not required to be in (however that is also a mistake, the goal is to keep one of each task type).

**Comment by KennethEnevoldsen** [comment]
Will merge this in the for the fixes to datasets - and then create an updated PR for the constructed benchmarks.

**Comment by KennethEnevoldsen** [comment]
Will merge this in for now it only makes minor changes to the repo - future PRs will added more on this

**Pull request merged**